Sources/MapboxNavigation/MapView.swift

```diff
@@ -174,33 +174,94 @@ extension MapView {
         
         return datasets
     }
-
-    var mainRouteLineParentLayerIdentifier: String? {
-        var parentLayer: String? = nil
-        let identifiers = [
+    
+    func layerPosition(for layerIdentifier: String, route: Route? = nil, customLayerPosition: LayerPosition? = nil) -> LayerPosition? {
+        guard customLayerPosition == nil else { return customLayerPosition }
+        
+        let belowSymbolLayers: [String] = [
+            route?.identifier(.routeCasing(isMainRoute: false)),
+            route?.identifier(.route(isMainRoute: false)),
+            route?.identifier(.routeCasing(isMainRoute: true)),
+            route?.identifier(.route(isMainRoute: true)),
+            route?.identifier(.restrictedRouteAreaRoute),
+            NavigationMapView.LayerIdentifier.arrowStrokeLayer,
             NavigationMapView.LayerIdentifier.arrowLayer,
-            NavigationMapView.LayerIdentifier.arrowSymbolLayer,
             NavigationMapView.LayerIdentifier.arrowSymbolCasingLayer,
-            NavigationMapView.LayerIdentifier.arrowStrokeLayer,
+            NavigationMapView.LayerIdentifier.arrowSymbolLayer
+        ].compactMap{ $0 }
+        let aboveSymbolLayers: [String] = [
+            NavigationMapView.LayerIdentifier.buildingExtrusionLayer,
```

Comment:
Should building extrusions occlude symbol layers? It would be helpful to see some screenshots of highlighted and unhighlighted buildings atop labels. My guess is that we could potentially get feedback either way, but on balance, it’s better to keep the labels visible than to occlude them.

Reply:
I see. I'll move the `buildingExtrusionLayer`  from the sequence. Because its previous layer position is above all the `building` layers. I'll remove the change of it.